source/Cosmos.System2_Plugs/System/ConsoleImpl.cs

```diff
@@ -429,17 +431,20 @@ public static ConsoleKeyInfo ReadKey(bool intercept)
             bool xControl = (key.Modifiers & ConsoleModifiers.Control) == ConsoleModifiers.Control;
 
             return new ConsoleKeyInfo(key.KeyChar, key.Key.ToConsoleKey(), xShift, xAlt, xControl);
-        }
-
+        }
+
+        private static List<char> chars = new List<char>(32);

```

Comment:
Why move chars outside the ReadLine function?

Reply:
because if we put them into the function, every thread would have to access its own instance of characters and this would just waste memory and cause errors when threads that don't require "keyboard read" are forced to add to that list(that for them doesn't exist) a character.